### PR TITLE
Add an option to specify the repository and branch name in dist-git p…

### LINKED ISSUE #147: Add an option to specify the repository and branch name in dist-git plugin
This could make it possible to automate the work on syncing. It could be done by extending the descriptor file with settings specific for this plugin, for example in the `dogen` section.

### PR DESCRIPTION
**Pull request by goldmann**
…lugin

This is implemneted by adding the --dist-git-repo and --dist-git-branch
switches to CLI. Additionally it's possible to specify repository and
branch name in the image descriptor.

```yaml
dist_git:
    repo: "your-repo-name"
    branch: "branch-name"
```

Fixes #147

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+dist
```
Comment (dbecvarik) [inline .gitignore]: why not dist-git -it will be easier to get what that dir is for

File: dogen/plugins/dist_git.py [hunk 2]
```diff
@@ -94,51 +133,34 @@ def stage_modified(self):
...
-        # Just check if the target branch is correct
-        if not Tools.decision("You are currently working on the '%s' branch, is this what you want?" % self.branch):
-            print("")
-            self.switch_branch()
+        if os.path.exists(self.output):
+            with Chdir(self.output):
+                self.log.info("Pulling latest changes in repo %s..." % self.repo)
+                subprocess.check_output(["git", "fetch"])
+                subprocess.check_output(["git", "reset", "--hard", "origin/%s" % self.branch])
+            self.log.debug("Done")
```
Comment (dbecvarik) [inline dogen/plugins/dist_git.py]: Done? I would prefer something like: "Cloned and prepared dist-git repository"

File: dogen/plugins/dist_git.py [hunk 3]
```diff
@@ -14,21 +15,58 @@ def info():
     def inject_args(parser):
         parser.add_argument('--dist-git-enable', action='store_true', help='Enables dist-git plugin')
         parser.add_argument('--dist-git-assume-yes', action='store_true', help='Skip interactive mode and answer all question with "yes"')
+        parser.add_argument('--dist-git-repo', help='Repository name to sync with')
+        parser.add_argument('--dist-git-branch', help='Branch name in the repository to sync with')
```
Comment (dbecvarik) [inline dogen/plugins/dist_git.py]: this is override for image.yaml? Is it really needed?  It can be used to use 1 image.yaml for two branches, which is weird use case.

File: dogen/plugins/dist_git.py [hunk 4]
```diff
@@ -76,13 +114,14 @@ def repo_info(path):
...
         return name, branch, commit
 
-    def __init__(self, log, source, path, noninteractive=False):
+    def __init__(self, log, output, source, repo, branch, noninteractive=False):
         self.log = log
-        self.path = path
-        self.dockerfile = os.path.join(self.path, "Dockerfile")
+        self.output = output
+        self.repo = repo
+        self.branch = branch
```
Comment (dbecvarik) [inline dogen/plugins/dist_git.py]: It seems you do not log branch name anywhere - i think it should be visible in log as it is pretty important part.

File: .gitignore (hunk 1 above)
Comment (goldmann) [inline .gitignore]: This is not related ;) It's a leftover from releasing to pypy.

File: dogen/plugins/dist_git.py (hunk 4 above)
Comment (goldmann) [inline dogen/plugins/dist_git.py]: Good idea!

File: dogen/plugins/dist_git.py (hunk 2 above)
Comment (goldmann) [inline dogen/plugins/dist_git.py]: I can extend it, sure.

File: dogen/plugins/dist_git.py (hunk 3 above)
Comment (goldmann) [inline dogen/plugins/dist_git.py]: It'll be removed.

File: dogen/plugins/dist_git.py [hunk 5]
```diff
@@ -18,17 +19,31 @@ def inject_args(parser):
...
+        self.branch = None
 
     def prepare(self, cfg):
         if not self.args.dist_git_enable:
             return
 
-        with Chdir(self.output):
-            self.git.prepare()
-            self.git.clean()
+        dist_git_cfg = cfg.get('dogen', {}).get('plugins', {}).get('dist_git', None)
```
Comment (goldmann) [inline dogen/plugins/dist_git.py]: Nice trick, huh?

Comment (dbecvarik) [inline dogen/plugins/dist_git.py]: PEP 20? :)
